### PR TITLE
add french translations to locales

### PR DESCRIPTION
**Pull request by coralieco**
Some french translations are missing:
image

If anyone could have a look and review 👍🏻

### CODE REVIEW

**Comment by dwradcliffe** [comment]
Thanks for the pull request! We usually like to get at least one review before merging translation changes. (I'm unqualified to review this 😄)

File: config/locales/fr.yml [hunk 1]
```diff
@@ -1,30 +1,30 @@
...
-  multifactor_authentication:
+  multifactor_authentication: Authentication Multifacteur
   subtitle: votre communauté d'hébergement des gems
   this_rubygem_could_not_be_found: Rubygem introuvable.
   time_ago: il y a %{duration}
   title: RubyGems.org
   update: Mise à jour
-  try_again:
-  advanced_search:
+  try_again: Une erreur est survenue. Veuillez réessayer.
```
Comment (benoittgt) [inline config/locales/fr.yml]: Sometimes we have dot sometimes not? Did you follow the eng local for dots too?

File: config/locales/fr.yml [hunk 2]
```diff
@@ -136,34 +136,34 @@ fr:
...
+      otp_prompt: Veuillez taper le code de l'application d'authentication pour poursuivre.
+      confirm: J'ai gardé mes codes de récupération en sécurité.
+      enable: Activer
+      account: "Compte: %{account}"
+      key: "Clé: %{key}"
       time_based:
     create:
-      qrcode_expired:      
-      success:
+      qrcode_expired: Le code QR et la clé sont expirés. Veuillez essayer d'enregistrer un nouvel appareil.
```
Comment (benoittgt) [inline config/locales/fr.yml]: I'm wondering if we should say "QR code", even if it's less french this is the way I heard it most of the time. 🤔

File: config/locales/fr.yml [hunk 3]
```diff
@@ -136,34 +136,34 @@ fr:
...
     new:
-      title:
-      scan_prompt:
-      otp_prompt:
-      confirm:
-      enable:
-      account:
-      key:
+      title: Activer l'authentication multifacteur
+      scan_prompt: Veuillez scanner le code QR avec votre app d'authentication. Si vous ne pouvez pas scanner le code, veuillez saisir manuellement les informations ci-dessous dans votre app.
```
Comment (benoittgt) [inline config/locales/fr.yml]: I'm wondering if we should say "QR code", even if it's less french this is the way I heard it most of the time. 🤔

File: config/locales/fr.yml [hunk 4]
```diff
@@ -177,29 +177,29 @@ fr:
         text: Besoin d'un nouveau mot de passe ? Pas de problème. %{link}
         title: Renvoi de mot de passe
       delete:
-        delete:
-        delete_profile:
-        warning:
+        delete: Supprimer
+        delete_profile: Supprimer le profil
+        warning: La suppression de votre profil est une action irréversible. Cela ne peut pas être annulé en contactant le support. Soyez sûr !
```
Comment (benoittgt) [inline config/locales/fr.yml]: "Soyez sur". Maybe "Soyez en sur"? 🤔

File: config/locales/fr.yml [hunk 5]
```diff
@@ -1,30 +1,30 @@
...
   help_url: http://help.rubygems.org
   locale_name: Français
   none:
-  not_found:
+  not_found: Introuvable
   please_sign_up: Accès refusé. Inscrivez-vous sur https://rubygems.org
   sign_in: Connexion
   sign_up: Inscription
-  multifactor_authentication:
+  multifactor_authentication: Authentication Multifacteur
```
Comment (thbar) [inline config/locales/fr.yml]: "Authentification" rather than "Authentication"?

File: config/locales/fr.yml [hunk 6]
```diff
@@ -251,27 +251,27 @@ fr:
     show_yanked:
       not_hosted_notice: Gem non hébergé sur Rubygems pour le moment.
       reserved_namespace:
-        one:
-        other:
+        one: Cette gem existait, mais a été supprimée par son propriétaire. L'équipe RubyGems.org a réservé ce nom de gem pour un jour supplémetaire. Une fois ce temps écoulé, quiconque pourra utiliser ce nom de gem en utilisant gem push. <br/> Si vous êtes le propriétaire précédent de cette gem, vous pouvez modifier la propriété de cette gem en utilisant la commande de gem owner. Vous pouvez également créer de nouvelles versions de cette gem avec gem push.
```
Comment (benoittgt) [inline config/locales/fr.yml]: Little typo "supplémentaire".

File: config/locales/fr.yml [hunk 7]
```diff
@@ -251,27 +251,27 @@ fr:
...
+        other: Cette gem existait, mais a été supprimée par son propriétaire. L'équipe RubyGems.org a réservé ce nom de gem pour %{count} jours supplémetaires. Une fois ce temps écoulé, quiconque pourra utiliser ce nom de gem en utilisant gem push. <br/> Si vous êtes le propriétaire précédent de cette gem, vous pouvez modifier la propriété de cette gem en utilisant la commande de gem owner. Vous pouvez également créer de nouvelles versions de cette gem avec gem push.
```
Comment (benoittgt) [inline config/locales/fr.yml]: Little typo "supplémentaire".

File: config/locales/fr.yml (hunk 5 above)
Comment (coralieco) [reply config/locales/fr.yml]: Oh yes ! Good catch, I will change it everywhere

File: config/locales/fr.yml (hunk 3 above)
Comment (coralieco) [reply config/locales/fr.yml]: I don't know, from Wikipédia: 
image

File: config/locales/fr.yml [hunk 8]
```diff
@@ -50,13 +50,13 @@ fr:
...
     create:
-      promise_resend:
+      promise_resend: Nous vous enverrons un email avec le lien de confirmation pour activer pour votre compte si vous en avez un.
     new:
-      submit:
-      title:
-      will_email_notice:
+      submit: Renvoyer la confirmation
+      title: Renvoyer l'email de confirmation
+      will_email_notice: Nous vous enverrons un email avec le lien de confirmation pour activer pour votre compte.
```
Comment (thbar) [inline config/locales/fr.yml]: I went from https://github.com/rubygems/rubygems.org/blob/master/config/locales/en.yml#L53 and used some inspiration from https://github.com/tigrish/devise-i18n/blob/master/rails/locales/fr.yml#L36.

I suggest "Si ce compte existe, vous recevrez un email avec un lien de confirmation pour l'activer."

File: config/locales/fr.yml [hunk 9]
```diff
@@ -95,27 +95,27 @@ fr:
...
+      link_expiration_explanation: Veuillez noter que ce lien n'est valide que pendant 15 minutes.
     deletion_complete:
-      subject:
-      body:
+      subject: Votre compte a bien été supprimé de RubyGems.org
+      body: "Votre demande de suppresion de compte sur RubyGems.org a bien été prise en compte. Vous pouvez toujours créer un nouveau compte sur la page %{sign_up}"
     deletion_failed:
-      subject:
-      body:
+      subject: Votre demande de suppresion de compte sur RubyGems.org a échoué
```
Comment (thbar) [inline config/locales/fr.yml]: Small typo here: "suppression" instead of "suppresion".

File: config/locales/fr.yml [hunk 10]
```diff
@@ -136,34 +136,34 @@ fr:
       title: Changement de mot de passe
       will_email_notice: Nous vous enverrons un email pour changer votre mot de passe.
   multifactor_auths:
-    incorrect_otp:
-    otp_code:
-    require_mfa_disabled:
-    require_mfa_enabled:
+    incorrect_otp: Votre clé OTP est incorrect.
```
Comment (thbar) [inline config/locales/fr.yml]: "incorrecte" rather than "incorrect".

File: config/locales/fr.yml (hunk 3 above)
Comment (thbar) [reply config/locales/fr.yml]: I don't know what's correct, but indeed I've never heard "code QR" in real life amongst french people (even non techies). That said I discover that many sites use "code QR" indeed!

**Comment by coralieco** [comment]
@thbar and @benoittgt I was also wondering if `gem` is feminine or masculine.
In the current french translations of this file it is used as a masculine noun. But I am using it as a feminine one (ex: "toutes les gems", "une gem")

File: config/locales/fr.yml (hunk 4 above)
Comment (thbar) [reply config/locales/fr.yml]: The [original](https://github.com/rubygems/rubygems.org/blob/master/config/locales/en.yml#L182) is "Please be certain!". Indeed "Soyez sûr" isn't exactly fluid. I'd maybe suggest something like "Réfléchissez-bien avant de continuer !" or another type of sentence!

File: config/locales/fr.yml [hunk 11]
```diff
@@ -177,29 +177,29 @@ fr:
...
+        delete_profile: Supprimer le profil
+        warning: La suppression de votre profil est une action irréversible. Cela ne peut pas être annulé en contactant le support. Soyez sûr !
       mfa:
-        multifactor_auth:
-        disabled:
-        go_settings:
-        enabled:
-        disable:
+        multifactor_auth: Authentication Multifacteur
+        disabled: Vous n'avez pas encore activer l'authentication multifacteur.
```
Comment (thbar) [inline config/locales/fr.yml]: "activé" instead of "activer"

File: config/locales/fr.yml [hunk 12]
```diff
@@ -177,29 +177,29 @@ fr:
...
     delete:
-      title:
-      confirm:
-      instructions:
-      list_only_owner:
-      list_multi_owner:
+      title: Supprimer le profil
+      confirm: Confirmer
+      instructions: Nous sommes désolés de vous voir partir. Veuillez entrer votre mot de passe dans le champ et confirmer.
+      list_only_owner: Ces gems seront effacées quand vous effacerez votre profil. Si vous désirez ajouter des propriétaires avant de supprimer votre profil, vous pouvez utiliser la commande %{command_link}.
```
Comment (thbar) [inline config/locales/fr.yml]: Woah do I understand that gems with a single owner will be deleted if the last owner deletes his/her account? I need to dive into this a bit further :smile:

**Comment by thbar** [comment]
I've made a first pass based solely on the English locale file + the PR, and suggested a few changes. I think it's mostly good, I'll let other go through this too!

**Comment by thbar** [comment]
> @thbar and @benoittgt I was also wondering if `gem` is feminine or masculine.
> In the current french translations of this file it is used as a masculine noun. But I am using it as a feminine one (ex: "toutes les gems", "une gem")

I've only seen it used as feminine around me, like you mention!

File: config/locales/fr.yml (hunk 3 above)
Comment (coralieco) [reply config/locales/fr.yml]: I am changing it to `QR code` as it is more used this way.

File: config/locales/fr.yml (hunk 12 above)
Comment (coralieco) [reply config/locales/fr.yml]: It seems like so 🤨
image

http://help.rubygems.org/discussions/questions/8961-delete-my-account

**Comment by benoittgt** [comment]
LGTM good work.

**Comment by sonalkr132** [comment]
@coralieco Can you please rebase it and squash your commits?

File: config/locales/fr.yml (hunk 12 above)
Comment (sonalkr132) [reply config/locales/fr.yml]: See also: https://github.com/rubygems/rubygems.org/pull/1301#discussion_r70223143

**Comment by coralieco** [comment]
Sorry, I forgot about the "squash your commits" part. I do it

**Comment by dwradcliffe** [comment]
Thanks everyone!

**Pull request merged**